devicetypes/qubino/qubino-flush-dimmer.src/qubino-flush-dimmer.groovy

```diff
@@ -0,0 +1,890 @@
+			log.debug "Preference ${it.key} has been updated from value: ${state.currentPreferencesState."$it.key".value} to ${settings."$it.key"}"
+			state.currentPreferencesState."$it.key".status = "syncPending"
+			if (it.type == "boolRange") {
+				def preferenceName = it.key + "Boolean"
+				if (notNullCheck(settings."$preferenceName")) {
+					if (!settings."$preferenceName") {
+						state.currentPreferencesState."$it.key".status = "disablePending"
+					} else if (state.currentPreferencesState."$it.key".status == "disabled") {
+						state.currentPreferencesState."$it.key".status = "syncPending"
+					}
+				} else {
+					state.currentPreferencesState."$it.key".status = "syncPending"
+				}
+			}
+		} else if (!state.currentPreferencesState."$it.key".value) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
+	// Preferences template end
+}
+
+private syncConfiguration() {
+	def commands = []
+	parameterMap.each {
+		if (state.currentPreferencesState."$it.key".status == "syncPending") {
+			commands += secure(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+			commands += secure(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		} else if (state.currentPreferencesState."$it.key".status == "disablePending") {
+			commands += secure(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+			commands += secure(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		}
+	}
+	sendHubCommand(commands)
+}
+
+def getCommandClassVersions() {
+	[
+			0x20: 1, // Basic Set
+			0x5A: 1, // Device Reset Locally
+			0x73: 1, // Power Level
+			0x98: 1, // Security
+			0x86: 2, // Version
+			0x72: 2, // Manufacturer Specific
+			0x27: 1, // Switch All
+			0x25: 1, // Switch Binary
+			0x26: 3, // Switch Multilevel
+			0x32: 3, // Meter
+			0x71: 1, // Alarm
+			0x85: 2, // Association
+			0x8E: 2, // Multi Channel Association
+			0x59: 2, // Association Group
+			0x70: 1, // Configuration
+			0x60: 3, // Multi Instance
+			0x31: 5, // Sensor MultiLevel
+			0x30: 2, // Sensor Binary
+			0x71: 3, // Notification
+			0xEF: 1, // Mark
+	]
+}
+
+def configure() {
+	def commands = []
+	commands << zwave.manufacturerSpecificV2.manufacturerSpecificGet().format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:1, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:2, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:3, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:4, nodeId:[zwaveHubNodeId]).format()
+	response(delayBetween(commands, 500))

```

Comment:
afaik wrapping in _response()_ is only necessary when commands are sent in, well, response to another incoming command